orchestrator/test_runner/src/upgrade.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{utils::*, happy_path_test, happy_path_test_v2};
+use clarity::Address as EthAddress;
+use deep_space::Contact;
+use gravity_proto::gravity::query_client::QueryClient as GravityQueryClient;
+use deep_space::client::ChainStatus;
+use tonic::transport::Channel;
+use web30::client::Web3;
+use gravity_proto::cosmos_sdk_proto::cosmos::bank::v1beta1::Metadata;
+use crate::ibc_metadata::submit_and_pass_ibc_metadata_proposal;
+
+/// Perform a series of integration tests to seed the system with data, then submit and pass a chain
+/// upgrade proposal
+/// NOTE: To run this test, use the tests/run-upgrade-test.sh command with an old binary, then in
+/// a separate terminal execute tests/run-tests.sh with V2_UPGRADE_PART_1 as the test type.
+/// After the test executes, you will need to wait for the chain to reach the halt height, which is
+/// output in a comment at the end of this test
+#[allow(clippy::too_many_arguments)]
+pub async fn v2_upgrade_part_1(
+    web30: &Web3,
+    contact: &Contact,
+    grpc_client: GravityQueryClient<Channel>,
+    keys: Vec<ValidatorKeys>,
+    gravity_address: EthAddress,
+    erc20_addresses: Vec<EthAddress>,
+) {
+    let metadata = footoken_metadata(contact).await;
+    submit_and_pass_ibc_metadata_proposal(metadata.name.clone(), metadata.clone(), contact, &keys).await;
+    run_all_recoverable_tests(web30, contact, grpc_client.clone(), keys.clone(), gravity_address,
+        erc20_addresses.clone(), metadata).await;
+
+    let curr_height = contact.get_chain_status().await.unwrap();
+    let curr_height = if let ChainStatus::Moving {block_height} = curr_height {
+        block_height
+    } else {
+        panic!("Chain is not moving!");
+    };
+    let upgrade_height = (curr_height + 50) as i64;
+    execute_upgrade_proposal(contact, &keys, None, UpgradeProposalParams {
+        upgrade_height,
+        plan_name: "mercury".to_string(),
```

Comment:
this seems to line up with the proposal here just fine
https://www.mintscan.io/gravity-bridge/proposals/27
https://www.mintscan.io/gravity-bridge/txs/1C7186E7519F8BE40AEB5026450281B88DAB0EF95C5620F8101230B8BCE1D3E3